client/commands.c

```diff
@@ -793,16 +766,40 @@ double xdagGetHashRate(void)
 	return g_xdag_extstats.hashrate_s / (1024 * 1024);
 }
 
-void read_command(char* buffer)
+int read_command(char *cmd)
+{	
+	char* line = linenoise("xdag> ");
+	if(line == NULL) return 0;
+	
+	if(strlen(line) > XDAG_COMMAND_MAX) {
+		printf("exceed max length");
+		strncpy(cmd, line, XDAG_COMMAND_MAX - 1);
+		cmd[XDAG_COMMAND_MAX-1] = '\0';
+	} else {
+		strcpy(cmd, line);
+	}
+	free(line);
+	
+	if(strlen(cmd) > 0) {
+		linenoiseHistoryAdd(cmd);
+		linenoiseHistorySave(COMMAND_HISTORY);
+	}
+	
+	return 0;
```

Comment:
Why you always return 0? You change returned type to int, but never use returned value.